src/entry-points/browser/extension.ts

```diff
@@ -15,15 +15,20 @@
  ********************************************************************************/
 
 import * as vscode from 'vscode';
-import { MemoryProvider } from '../memory-provider';
-import { MemoryWebview } from '../views/memory-webview-main';
+import { AdapterRegistry } from '../../plugin/adapter-registry/adapter-registry';
+import { GdbCapabilities } from '../../plugin/adapter-registry/gdb-capabilities';
+import { MemoryProvider } from '../../plugin/memory-provider';
+import { MemoryWebview } from '../../plugin/memory-webview-main';
 
-export const activate = async (context: vscode.ExtensionContext): Promise<void> => {
+export const activate = async (context: vscode.ExtensionContext): Promise<AdapterRegistry> => {
```

Comment:
If we are returning an API, it could be good to publish the API to npm (or even on the GitHub npm registry).

We do this internally by having an api folder with the interface exposed (which is published as a separate package) and then have the main class implement this interface.

Reply:
I think that's a good idea, since it will make it easier to inform people of changes. On the one hand, I think that allowing others to contribute handling for their own adapters is valuable, on the other hand managing the API when VSCode's dependency resolution is fairly crude could be tricky. I know that we would like to support at least memory watchpoints on the plugin side - is there anything that falls on the margins of the DAP that you'd like to support? Perhaps we can start releasing the API once we've got at least a first draft of the functionality we know we want?

Reply:
Using our browser debugger and the gdb debugger as test adapters should really help us uncover the correct abstraction points and any common code we don't need to duplicate in plugins. Let's let this evolve as we add features and see where it goes :)